File: app/src/main/java/net/wojteksz128/worktimemeasureapp/notification/IgnoreReminderAction.java
package net.wojteksz128.worktimemeasureapp.notification;

import android.content.Context;
import android.util.Log;

class IgnoreReminderAction implements NotificationAction {

    private static final String LOG = IgnoreReminderAction.class.getSimpleName();

    @Override
    public void invoke(Context context) {
        Log.d(LOG, "invoke: Ignore notification clicked");
        NotificationUtils.clearAllNotifications(context);
    }
}

File: app/src/main/kotlin/net/wojteksz128/worktimemeasureapp/notification/action/IgnoreReminderAction.kt
package net.wojteksz128.worktimemeasureapp.notification.action

import android.content.Context
import android.util.Log

import net.wojteksz128.worktimemeasureapp.util.notification.NotificationUtils

internal object IgnoreReminderAction : NotificationAction {

    private val LOG = IgnoreReminderAction::class.java.simpleName

    override operator fun invoke(context: Context) {
        Log.d(LOG, "invoke: Ignore notification clicked")
        NotificationUtils.clearAllNotifications(context)
    }
}
